src/machine/machine-dbus.c

```diff
@@ -1097,6 +1099,14 @@ int bus_machine_method_copy(sd_bus_message *message, void *userdata, sd_bus_erro
         if (r == 0)
                 return 1; /* Will call us back */
 
+        r = machine_get_uid_gid_base(m, &uid);
+        if (r < 0)
+                return r;
+
+        /* Is user namespacing is used? ifso set the resulting UIDs always to zero */
+        if (uid != 0)
+                copy_flags |= COPY_CHOWN_ROOT;
```

Comment:
uh, it didn't! I only tested copy from the container, not copy to the container, hence I didn't notice. I reworked this now, to fix it up in both directions!